packages/dds/sequence/src/test/reentrancy.spec.ts

```diff
@@ -114,6 +118,70 @@ describe("SharedString op-reentrancy", () => {
 			}
 		});
 
+		it("is empty after deleting reference pos in reentrant callback", () => {
+			sharedString.insertText(0, "abcX");
+			const { segment } = sharedString.getContainingSegment(0);
+			assert(segment);
+			segment.localRefs ??= new LocalReferenceCollection(segment);
+			const localRef = segment.localRefs.createLocalRef(
```

Comment:
could you add assertions to each test about the final state of `localRef.getSegment()` and `localRef.getOffset()`?